rmw_fastrtps_shared_cpp/src/custom_publisher_info.cpp

```diff
@@ -114,24 +114,30 @@ void PubListener::set_on_new_event_callback(
     switch (event_type) {
       case RMW_EVENT_LIVELINESS_LOST:
         publisher_info_->data_writer_->get_liveliness_lost_status(liveliness_lost_status_);
-        callback(user_data, liveliness_lost_status_.total_count_change);
-        liveliness_lost_status_.total_count_change = 0;
+        if (liveliness_lost_status_.total_count_change) {
```

Comment:
nitpick: prefer explicitly checking for greater than 0 rather than an implicit conversion to boolean
```
if (liveliness_lost_status_.total_count_change > 0)
```